### PR TITLE
sys-fs/eudev: add 3.2.14

### PR DESCRIPTION
**Pull request by orbea**
This will allow libgudev to build with eudev and should likely work in most if not all cases.

A friend who uses eudev and libgudev tested it without issues against the following list of packages:
```
net-libs/libqmi-1.30.8:0/5.8::gentoo  USE="mbim qrtr -gtk-doc"
sys-fs/udisks-2.9.4-r3:2::gentoo  USE="acl daemon elogind introspection lvm nls (selinux) -debug -systemd -vdo -zram"
sys-power/upower-0.99.20:0/3::gentoo  USE="introspection (selinux) -doc -ios -test"
dev-libs/libwacom-2.8.0:0/9::gentoo  USE="-doc -test"
media-gfx/gimp-2.10.34-r2:0/2::gentoo  USE="alsa udev -aalib (-aqua) -debug -doc -gnome -heif -jpeg2k -jpegxl -mng -openexr -postscript -test -unwind -vector-icons -webp -wmf -xpm" CPU_FLAGS_X86="mmx sse"
sys-power/switcheroo-control-2.6-r2::gentoo  USE="-gtk-doc -test" PYTHON_SINGLE_TARGET="python3_11 -python3_10"
gnome-base/gvfs-1.50.6::gentoo  USE="elogind http policykit udev udisks -afp -archive -bluray -cdda -fuse -gnome-online-accounts -google -gphoto2 -ios -keyring -mtp -nfs -samba -systemd -test -zeroconf"
net-misc/modemmanager-1.18.12:0/1::gentoo  USE="elogind introspection policykit qmi qrtr (selinux) udev -mbim -systemd -test -vala"
sys-power/power-profiles-daemon-0.13::gentoo  USE="-gtk-doc -test" PYTHON_SINGLE_TARGET="python3_11 -python3_10"
xfce-base/thunar-4.18.7:0/3::gentoo  USE="pcre trash-panel-plugin udisks -exif -introspection -libnotify"
xfce-base/thunar-volman-4.18.0::gentoo  USE="-libnotify"
```
Upstream-Issue: https://github.com/eudev-project/eudev/issues/249
Upstream-PR: https://github.com/eudev-project/eudev/pull/253
Upstream-Commit: https://github.com/eudev-project/eudev/commit/c5bae0b656513463f92808f324f8fcbe34a0b401

### CODE REVIEW

File: virtual/libudev/libudev-251-r2.ebuild [hunk 1]
```diff
@@ -12,14 +12,16 @@ KEYWORDS="~alpha amd64 arm arm64 hppa ~ia64 ~loong ~m68k ~mips ppc ppc64 ~riscv
...
-# eudev does not provide v251 APIs, see
-# https://github.com/eudev-project/eudev/issues/249
 RDEPEND="
 	!systemd? (
-		eudev? ( sys-fs/eudev[${MULTILIB_USEDEP}] )
+		eudev? (
+			sticky-tags? (
+				>=sys-fs/eudev-3.2.12-r2[${MULTILIB_USEDEP}]
+			)
+			sys-fs/eudev[${MULTILIB_USEDEP}]
```
Comment (floppym) [inline virtual/libudev/libudev-251-r2.ebuild:23]: The sticky-tags USE flag was meant as temporary hack.

Please drop the sticky-tags USE flag and depend >=sys-fs/eudev-3.2.12-r2 unconditionally.

**Review by floppym — CHANGES_REQUESTED** [review]
I don't support merging the eudev change without upstream actually merging it and without a Gentoo developer willing to actually maintain the ebuild.

I do have a change request for virtual/libudev.

File: virtual/libudev/libudev-251-r2.ebuild (hunk 1 above)
Comment (orbea) [reply virtual/libudev/libudev-251-r2.ebuild]: Sure, I changed it.

**Comment by orbea** [comment]
> I don't support merging the eudev change without upstream actually merging it and without a Gentoo developer willing to actually maintain the ebuild.

If given the chance I am willing to maintain it until a better option emerges or if and when it becomes impossible to do that.

I also fully understand the concern about merging a patch upstream eudev has not merged, but my understanding is the lack of testing so this may be the best way to do that. At least is a improvement in that it allows libgudev to build.

**Comment by orbea** [comment]
@floppym I sincerely still do not understand why there is so much resistance to me maintaining eudev within ::gentoo? I also find it extremely disconcerting how rude SOME (Not all) of the Gentoo devs were when this was brought up on the ML.

**Comment by KenjiBrown** [comment]
I have pushed eudev-3.2.12-r2.ebuild (with the stickytags patch) to without-systemd overlay
https://github.com/KenjiBrown/without-systemd/blob/master/sys-fs/eudev/eudev-3.2.12-r2.ebuild
After the update I have been able to compile dev-libs/libgudev-238-r1::gentoo without any other problem.

**Comment by orbea** [comment]
> I don't support merging the eudev change without upstream actually merging it

This is now merged upstream.

https://github.com/eudev-project/eudev/commit/c5bae0b656513463f92808f324f8fcbe34a0b401

**Comment by orbea** [comment]
@floppym The way you just ignored this is really inconsiderate.

**Comment by floppym** [comment]
@orbea Not sure what you expected from me. I have no interest in proxying commits for this package.

**Comment by SoapGentoo** [comment]
@orbea we have told you eudev isn't coming back. What is so hard to understand about that?

**Comment by floppym** [comment]
A heads-up: I just dropped eudev from virtual/libudev and virtual/udev in gentoo. Anyone maintaining eudev in an overlay will likely need to provide replacements for those virtuals as well.

**Pull request closed without merging**